src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,301 @@
+import React,{useCallback,useMemo,useRef,forwardRef,useState,useLayoutEffect} from 'react'
+import styled from 'styled-components'
+import { Mixins } from '../../Utils'
+import {
+    Main,
+    MainInterface,
+} from '../../Utils/BaseStyles';
+import Theme from '../../Themes/ThemeTemplate'
+import {Input} from '../../Inputs/Input/Input'
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+const DECIMAL_BASIS=10
+const PERCENTAGE_FACTOR=100
+
+interface IAdditionalProps{
+    /* if the heading entry must to be taken as the reference entry for taking width from */
+    ref?:boolean;
```

Comment:
Pick One Entry to Specify as the Max Width of the Component.
isMaxWidthAdjustedtoThisComponent